### DISCUSSION TITLE
Getting this error 'Error (1): fast_xml_parser_1.XMLParser is not a constructor'

### DISCUSSION THREAD

**Discussion by vinayahercsa** [NaturalIntelligence/fast-xml-parser#563] [Q&A]
I am getting 'Error (1): fast_xml_parser_1.XMLParser is not a constructor' while running the Jenkins job for components deployement into salesforce org using 'sfdx force:source:deploy  -x manifest/package.xml -u ${sf_username}',as a week before the deployement was working fine can anyone suggest me something.

**Comment by amitguptagwl** [comment]
Are you using the fast XML Parser directly? or as dependency of other project? Can you please let us know the version and sample code?

**Reply by vinayahercsa** [reply]
#!groovy
import groovy.json.JsonSlurperClassic
import io.jenkins.plugins.*
import jenkins.model.*

node {

    
    
         def toolbelt = tool 'sfdx'
         def JWT_KEY_CRED_ID=env.JWT_KEY_CRED_ID
def clientId = env.CLIENT_ID
def sf_username=env.sf_username
        
 stage('checkout source') {
        // when running in multi-branch job, one must issue this command
        checkout scm
    }

    withCredentials([file(credentialsId: JWT_KEY_CRED_ID, variable: 'jwt_key_file')]) {
        stage('Deploye Code') {
            if (isUnix()) {
                rc = sh returnStatus: true, script: "${toolbelt}/sfdx force:auth:jwt:grant --clientid ${clientId} --jwtkeyfile ${jwt_key_file} --username ${sf_username} --instanceurl https://test.salesforce.com --setdefaultdevhubusername"
                				
        }else{
                rc = sh returnStatus: true, script: "${toolbelt}/sfdx force:auth:jwt:grant --clientid ${clientId} --jwtkeyfile ${jwt_key_file} --username ${sf_username} --instanceurl https://test.salesforce.com --setdefaultdevhubusername "
            }
            if (rc != 0) { error 'hub org authorization failed' }

			println rc}
			
			// need to pull out assigned username  force:source:deploy -x path/to/package.xml
			if (isUnix()) {
                sh 'ls -la'

				rmsg = sh returnStdout: true, script: "${toolbelt}/sfdx force:source:deploy --checkonly -u ${sf_username} -x manifest/package.xml"


			}else{
			   rmsg = bat returnStdout: true, script: "${toolbelt}/sfdx  force:source:deploy -checkonly manifest/package.xml  -u ${sf_username}"
			}
       
    }
		 	  
    
       
        
       
        
    
        }

**Reply by vinayahercsa** [reply]
@amitguptagwl  In the above Script for sfdx force:source:deploy it is giving this error

**Comment by amitguptagwl** [comment]
I still didn't get the answer what version of fast-xml-parser you're using and how. I'm not able to understand how this script is related to fast-xml-parser

**Reply by vinayahercsa** [reply]
yes that is the issue i am not using fast-xml-parser anywhere in my sfdx project but still getting this error.

**Reply by amitguptagwl** [reply]
there might be a possibility that some of the dependency is using FXP> But I can't help in that, I guess. You'll have to debug more to find and report it.
